projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
 # Entity Management
 
-**UNDER CONSTRUCTION**
+`withEntities` is an extension to facilitate CRUD operations for managing entities. It shares similarities with @ngrx/entity but is part of the Signal Store.
+
+It doesn't include methods for backend communication. You have to implement them separately.
+
+## Creating the Store
+
+A store implementing a `Todo` entity can have the following implementation:
+
+```typescript
+interface Todo {
+  id: number;
+  name: string;
+  description: string;
+}
+
+const TodoStore = signalStore(withEntities<Todo>());
+```
+
+`withEntities` adds three properties of type `Signal` to the `TodoStore`.
+
+- `ids: Signal<EntityId[]>`: ids of all entities
+- `entities: Signal<Todo[]>`: array of all entities
+- `entityMap: Signal<EntityMap<Todo>>`> map of entities where the key is the id (`EntityId`)
+
+By default, `withEntities` requires your entity to have a property of name `id`, which serves as a unique identifier. `id` has to be of type `EntityId`, which is a `number` or a `string`.
+
+## Adding & Setting
+
+`addEntity` and `addEntities` add entries to the store. They act as updater functions for `patchState`.
+
+Here is an example on how to use them inside a component.
+
+```typescript
+@Component({
+  template: `@for (todo of todoStore.entities();track todo.id) {
+    <ul>
+      <li>{{ todo.name }}</li>
+    </ul>
+    }`,
+  standalone: true,
+  providers: [TodoStore],
+})
+export class EntityComponent implements OnInit {
+  todoStore = inject(TodoStore);
+
+  ngOnInit() {
+    // add a single entity
+    patchState(this.todoStore, addEntity({ id: 1, name: 'Car Washing', finished: false }));
+
+    // add multipe entities
```

Comment:
```suggestion
    // add multiple entities
```

Reply:
Thanks!

Reply:
Sorry @swami-sanapathi , had to revert. Was not aware that I could add you as co-contributor. Now it is how it should be.